scripts/release.js

```diff
@@ -0,0 +1,130 @@
+/*!
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+require('shelljs/global');
+
+var semver = require('semver');
+var path = require('path');
+var fs = require('fs');
+var helpers = require('./helpers');
+
+var Module = helpers.Module;
+var run = helpers.run;
+var git = helpers.git;
+
+var TRAVIS_TAG = process.env.TRAVIS_TAG;
+var MODULE_NAME;
+var MODULE_VERSION;
+
+if (semver.valid(TRAVIS_TAG)) {
+  MODULE_NAME = Module.UMBRELLA;
+  MODULE_VERSION = TRAVIS_TAG;
+} else {
+  var parts = TRAVIS_TAG.split('-');
+
+  MODULE_VERSION = parts.pop();
+  MODULE_NAME = parts.pop();
+}
+
+var mod = new Module(MODULE_NAME);
+
+mod.install();
+mod.runSystemTests();
```

Comment:
> does not test to see if the release integrates with other modules

A change to common shouldn't run other modules' tests because common changes have to come first before the other modules update their dependency on common. We have to trust the integrity of our `common` test suite to know that we aren't breaking anything.

> google-cloud does not run tests for any other modules (PubSub, BigQuery, etc.)

If our dev->release process goes right, it shouldn't have to. Only modules that pass unit & system tests should be released, so google-cloud would never release a non-working module.

Am I not thinking of something where this wouldn't work?